src/test/java/teammates/ui/webapi/UpdateFeedbackSessionActionTest.java

```diff
@@ -94,6 +99,216 @@ protected void testExecute() {
+        FeedbackSessionData response = (FeedbackSessionData) r.getOutput();
+
+        Map<String, Long> expectedInstructorDeadlines = new HashMap<>();
+        expectedInstructorDeadlines.put("instructor1@course1.tmt", 1809126000000L);
+        expectedInstructorDeadlines.put("instructor2@course1.tmt", 1809126000000L);
+        expectedInstructorDeadlines.put("helper@course1.tmt", 1809126000000L);
+        assertEquals(expectedInstructorDeadlines, response.getInstructorDeadlines());
+
+        ______TS("update deadline extension for instructor");
+
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        newInstructorDeadlines = updateRequest.getInstructorDeadlines()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toEpochMilli()));
+        newInstructorDeadlines.put("helper@course1.tmt", 1809298800000L);
+        updateRequest.setInstructorDeadlines(newInstructorDeadlines);
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        expectedInstructorDeadlines = new HashMap<>();
+        expectedInstructorDeadlines.put("instructor1@course1.tmt", 1809126000000L);
+        expectedInstructorDeadlines.put("instructor2@course1.tmt", 1809126000000L);
+        expectedInstructorDeadlines.put("helper@course1.tmt", 1809298800000L);
+        assertEquals(expectedInstructorDeadlines, response.getInstructorDeadlines());
+
+        ______TS("delete deadline extension for instructor");
+
+        // the typical update request does not contain the course 1 helper instructor's email
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+
+        a = getAction(updateRequest, param);
+        r = getJsonResult(a);
+        response = (FeedbackSessionData) r.getOutput();
+
+        expectedInstructorDeadlines = new HashMap<>();
+        expectedInstructorDeadlines.put("instructor1@course1.tmt", 1809126000000L);
+        expectedInstructorDeadlines.put("instructor2@course1.tmt", 1809126000000L);
+        assertEquals(expectedInstructorDeadlines, response.getInstructorDeadlines());
+
+        ______TS("change deadline extension for non-existent instructor; "
+                + "should throw EntityNotFoundException");
+
+        updateRequest = getTypicalFeedbackSessionUpdateRequest();
+        newInstructorDeadlines = updateRequest.getInstructorDeadlines()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().toEpochMilli()));
+        newInstructorDeadlines.put("nonExistentInstructor@gmail.tmt", 1809126000000L);
+        updateRequest.setInstructorDeadlines(newInstructorDeadlines);
+
+        a = getAction(updateRequest, param);
+        final UpdateFeedbackSessionAction a1 = a;
+        assertThrows(EntityNotFoundException.class, a1::execute);
```

Comment:
I don't believe this needs to be effectively final if you are using method reference. 

Might be wrong though. If I am wrong, I think something like this would be better:

```suggestion
        UpdateFeedbackSessionAction a1 = getAction(updateRequest, param);
        assertThrows(EntityNotFoundException.class, a1::execute);
```

Reply:
Yea, you're right. I started with the lambda expressions initially but I changed it. I forgot I could remove the final fields. Thanks.